### DISCUSSION TITLE
Reference pool members doesn’t get reward after changes pool’s wallets

### DISCUSSION THREAD

**Discussion by FreakyHead** [Chia-Network/chia-blockchain#8519] [Q&A] [answered]
We create a pool based on the reference pool.

Everything were working as expected and we were testing by the 10 members.
We changed pool’s wallets (wallet for pool’s commissions and wallet for distribution) and members were stop receiving any reward after blocks were find.

It’s very strange because members who’s find the blocks still get the 0,25 coin. But on pool’s wallets we doesn’t get any coins. This coins still sent to previous wallets.

**After we changed wallets back, all start work as should**

Also we were starting got warnings in pool log. Such as (ids were changed):

```
INFO     Is 806fe02b0d1cfdd83523e641a2ad3d5762c80212db6f61...... pool member: False
INFO     Singleton is not assigned to this pool
INFO     Wrong target puzzle hash: e3286a22a66a9f78ad7cdee409c8b3f05820ce2088e3284226ba48b96...... for launcher_id 0cf19eaa6600153db23433fc51a98060f319a9128e07f509.....
```

I have a few questions:

What initial reason may happen of this behavior?
What we should doing after changing pool wallets?
What initial reason may happen of this behavior?

I have been thinking about such reason: After changing pool’s wallets all member should recreate NFT for pool. But I’m not sure

**Comment by mariano54** [comment, marked as answer]
The NFT that is created, is created towards a specific pool address, so you cannot change that address.
If you decide to use a new address, you will have to keep supporting the old one for all of your original pool members.
The address itself is hardcoded onto the singleton in the blockchain, and changing it requires the user to change pools.

**Reply by FreakyHead** [reply]
Can pool member leave pool and join again after address will changed?
Did he need recreate plots again? Or maybe he may recreate only NFT?

**Reply by FreakyHead** [reply]
> If you decide to use a new address, you will have to keep supporting the old one for all of your original pool members.

Pool has only a one address in config at time (one for pool's fee and second for distribution) how I can to keep 2 addresses?

**Reply by FreakyHead** [reply]
@mariano54 please provide a solution or detail clarification for solve this

**Reply by mariano54** [reply]
Pool members can leave and join again yes, as long as you changed the details in the pool-info HTTP response.
The pool reference code is meant as an example. If you want to support more than one pool puzzle hash, you should make changes to it to support that.

**Reply by FreakyHead** [reply]
@mariano54 Ok. I won't want to implement functionality for using few puzzles. Only once modify address. Because all works fine on the previous. 

After leave the pool member should recreate nft? Created plots before will be accepted with new nft?

**Comment by mariano54** [comment]
you do not need to create new plot NFTs. just change from the pool to another pool, and then back to yours, or just use a different url in your pool (recommended)

**Reply by FreakyHead** [reply]
@mariano54 When you says address you mean pool’s wallet addresses or pool url? I hope wallet’s address)

**Reply by FreakyHead** [reply]
@mariano54 can you answer on last message?
